src/Agoda.Analyzers/AgodaCustom/AG0019PreventUseOfInternalsVisibleToAttribute.cs

```diff
@@ -45,11 +43,11 @@ private void AnalyzeNode(SyntaxNodeAnalysisContext context)
         {
             var attributeNode = (AttributeListSyntax)context.Node;
 
-           if (attributeNode.Target?.Identifier.Text != "assembly" || attributeNode.Attributes.Count == 0) return;
+           if (attributeNode.Target?.Identifier.Text != "assembly") return;
            
             foreach (var attribute in attributeNode.Attributes)
             {
-                if (attribute.Name != null && ((IdentifierNameSyntax)attribute.Name).Identifier.Text == "InternalsVisibleTo")
+                if (attribute.Name != null && (attribute.Name as IdentifierNameSyntax).Identifier.Text == "InternalsVisibleTo")
```

Comment:
`if (attribute.Name is IdentifierNameSyntax name && name.Identifier.Text == "InternalsVisibleTo")` must be best approach.

Reply:
Updated. Thank you.